### DISCUSSION TITLE
TimeCode not being written to the PCM (raw audio) files

### DISCUSSION THREAD

**Discussion by RobertMarkEnger** [gopro/labs#741] [Feature Requests]
I am running a GoPro11, with the MediaMod.  I have the Labs software installed, ver 02.12.70.
I am using QR code to jam-sync timecode into the GoPro.
The MP4 file  DOES contain timecode info.
BUT, **the PCM-wav raw audio file does NOT contain any timecode info**.
This is a shame, as the minimal (no processing) PCM audio files sound GREAT.  You get real audio dynamics (loud and soft passages) which add vitality to the video.  But, there is no easy way to import this for video editing.  

(Can't very well use Clapper-board for sound sync.  And the audio sources (car engines and drivers talking with helmets on) don't lend themselves to easy visual synchronization.  We can only sync based on events like car collisions, which may take place at time well into the edit segment.  This seems to make it too difficult to leverage the PCM files.)

![GoProLABS_wav_file_no_timecode](https://github.com/gopro/labs/assets/157267205/d7a892fa-e339-4eec-8e62-647681daeb85)
![GoProLABS_MP4_file_inspection-by-MediaInfo-tool](https://github.com/gopro/labs/assets/157267205/f1919181-57fc-4ba3-a3e6-b59d91856762)
![GoProLABS_wav_file_inspection-by-MediaInfo-tool](https://github.com/gopro/labs/assets/157267205/02fb332d-e69b-40e5-902f-f439f3cbd569)

**Comment by dnewman-gpsw** [comment]
This does seem to be an omission that the WAV files don't have timecode in it header.  That should be in the main firmware with it stock timecode support (there since HERO3 in MP4s.)  I don't think this is something Labs can easily add, although I will look into what the WAV formatting requires for timecode. 

PS. Labs for HERO11 is now on 2.20.70

**Reply by RobertMarkEnger** [reply]
hi:
I am a newcomer to GoPro, and am not sure I fully understand the breakdown of responsibilities between the "stock" and "labs" groups.
If possible, could someone from this "labs" community refer the WAV FILE TIMECODE issue over to the "stock software" group?
(Seems likely that Project-management will respond to a suggestion from the LABS group.  While small customers have a small voice...)
It would be very helpful to have the WAV files contain timecode!  Thank you.

**Reply by dnewman-gpsw** [reply]
Timecode in MP4 is definitely "stock" firmware, particular as QR Code jam-sync is a mainline feature on HERO12.  Timecode in WAV might be consider a feature for WAV for HERO12, likely not for HERO11.  Labs could consider added support. Can you please share a .WAV with timecode.  None of my audio devices produce timecode.

**Reply by RobertMarkEnger** [reply]
Hello David:
I have a Zoom-H6 and a newer M4-MicTrak
I am supplying test files from both of them.  BUT, they are not related.  I just grabbed a TentacleSync E mk-2 and set it to cellphone time.  I even rebooted it between tests.  So the TC from the H6 and the M4 tests are unrelated.

Re the Zoom-H6:
It has time-code, but based on its internal clock, and that clock cannot be disciplined.  What we do is record LTC on an audio channel, and then post-process using TentacleSync's TimeCodeTool.  When the tool is able to recognize the files and do things correctly, it will read the LTC on the audio channel, figure out the correct starting time code, and then write that time into the other related audio files that are fed into the tool.    So in this case, I recorded audio using the H6's XY mics as a stereo track, and put LTC on Tr1.  I then ran those files into TimeCodeTool.

I am supplying three files from the H6 test.  

- ZoomH6_tr1.wav                     is the LTC channel (just for reference).  
- ZoomH6_direct_LR.wav           is the audio file straight from the H6 (before being run through TimeCodeTool).  
- ZoomH6_TimeCodeTool_LR    is the audio file AFTER processing by TimeCodeTool.

Re the Zoom-M4mictrak
It accepts external timecode signal.  (Either to jam, or to operate "external" where the internal oscillator continuously chases the external TC being given to it.  I ran using the EXTERNAL mode.

One file is being supplied for the M4 test:

- ZoomM4_001_TrMic.wav         it is the file directly output by the M4.  It contains TC stamp (as well as some other information tags)

Hope these files are of use.   I hope the main-product "stock firmware" folks will listen to the Labs folks, and add TC to the wav files.
(Would also be nice if they modernized to BWF64 while they are at it.)

NOTE:  APPARENTLY THE GITHUB CHAT SYSTEM WILL NOT ALLOW DIRECT ATTACHMENT OF WAV FILES?
It says it will support ZIP.  So I will zip-up the files and submit that as a single ZIP file.
[WAV_TC_Samples_for_GoPro-Labs.zip](https://github.com/gopro/labs/files/14831032/WAV_TC_Samples_for_GoPro-Labs.zip)

Thank you again for looking at this issue.
Bob Enger

**Reply by dnewman-gpsw** [reply]
I have an experimental build to try if you are interested.

**Comment by RobertMarkEnger** [comment]
David:
Sorry for the delay.  I would like to try the new software.  Will this be a
version of LABS software, supporting its existing extensions?

Please let me know how to proceed.

Thank you!
Bob Enger

**Reply by dnewman-gpsw** [reply]
Yes, same features just not QA'd https://www.dropbox.com/scl/fi/bmog01uyke3uxhewbgtkf/HERO11Labs023071-Apr7.zip?rlkey=ehispt5jhr4vs9lhlj25ulo6q&dl=1

**Comment by RobertMarkEnger** [comment]
David:
I apologize for the delay.  The special modification places time-stamps on
the PCM files, and those timestamps agree with the timestamp on the video
file!  Examining sound-sync (using Davinci Resolve) shows they are indeed
synchronized.
THANK YOU!
1)  Can we get a version that works on the GoPro Hero 12?  (I now own 3 of
those)
2)  Can this special mod be included into the "main" LABS software, so that
any future versions of LAB software will continue to put timestamps on the
WAV files?
3)  Will this get transferred over to the folks who maintain the regular
(non-LABS) GoPro production software, so they will put it into the main
product?

Thank you!
Bob Enger

**Reply by dnewman-gpsw** [reply]
1 & 2) It was added into Labs core code, so it should be in the next HERO12 release (maybe some future models.)
3) No. Certainly not a direct port from Labs. Labs is used to model the demand for correctly implemented features (with all the QA implementations). This is not an in-demand feature, as you are the first and only to ask, so it will remain in Labs.

**Comment by RobertMarkEnger** [comment]
Thank you for making it available!

Is there a SW version number I should look for when getting LABS software
for the Hero-12?
(Is there an expected release date?)

**Reply by dnewman-gpsw** [reply]
Maybe in two months or so.  Nothing pending.

**Comment by RobertMarkEnger** [comment]
Thank you again!!
